### PR TITLE
science/orthanc: Revive: DICOM server for healthcare and medical research

### PR DESCRIPTION
**Pull request by yzgyyang**
PR:		242546
Approved by:	lwhsu (mentor)

### CODE REVIEW

File: science/orthanc/Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Created by: maintainer.freebsd@xpoundit.com
+
+PORTNAME=	orthanc
+DISTVERSION=	1.9.3
+CATEGORIES=	science
+MASTER_SITES=	https://www.orthanc-server.com/downloads/get.php?path=/orthanc/
+DISTNAME=	Orthanc-${PORTVERSION}
+
+MAINTAINER=	maintainer.freebsd@xpoundit.com
+COMMENT=	Orthanc is a DICOM server for healthcare and medical research
```
Comment (lwhsu) [inline science/orthanc/Makefile]: I think we can drop "Orthanc is a" part.

**Comment by lwhsu** [comment]
Don't forget encourage submitter to update the link at https://www.orthanc-server.com/download.php

File: science/orthanc/Makefile (hunk 1 above)
Comment (lwhsu) [reply science/orthanc/Makefile]: The old COMMENT was `RESTful DICOM server for healthcare and medical research`
https://svnweb.freebsd.org/ports/head/science/orthanc/Makefile?revision=471770&view=markup&pathrev=471893

File: science/orthanc/Makefile [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Created by: maintainer.freebsd@xpoundit.com
```
Comment (lwhsu) [inline science/orthanc/Makefile:1]: I somehow don't know if we should keep this created by line before it was deleted

https://svnweb.freebsd.org/ports/head/science/orthanc/Makefile?revision=471770&view=markup&pathrev=471893#l1

Or this is also fine.

I'm OK for both, maybe the new maintainer revives this is better

File: science/orthanc/files/orthanc.in [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+# orthanc_enable (bool):	Set to NO by default.
+#				Set it to YES to enable Orthanc.
+# orthanc_flags (str):		Set to %%ETCDIR%%/orthanc.json
+#				by default.
+
+. /etc/rc.subr
+
+name=orthanc
+rcvar=orthanc_enable
+desc="RESTful DICOM server for healthcare and medical research"
```
Comment (lwhsu) [inline science/orthanc/files/orthanc.in]: Should we align with the COMMENT in Makefile?

**Comment by lwhsu** [comment]
> Don't forget encourage submitter to update the link at https://www.orthanc-server.com/download.php

And try to upstream the local patches.

**Review by lwhsu — APPROVED** [review]

File: science/orthanc/Makefile (hunk 2 above)
Comment (yzgyyang) [reply science/orthanc/Makefile:1]: I have no strong feelings, but slightly prefer the current maintainer, so I will allow it ;)

File: science/orthanc/Makefile [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		USE_SYSTEM_LIBJPEG USE_SYSTEM_LIBPNG USE_SYSTEM_LUA \
+		USE_SYSTEM_OPENSSL USE_SYSTEM_PUGIXML USE_SYSTEM_SQLITE \
+		USE_SYSTEM_UUID USE_SYSTEM_ZLIB
+CFLAGS+=	-DNDEBUG
+CXXFLAGS+=	-DNDEBUG
+
+USE_RC_SUBR=	orthanc
+
+USERS=		orthanc
+GROUPS=		orthanc
```
Comment (lwhsu) [inline science/orthanc/Makefile:46]: Note that when defining this, we also need to update ports/{UIDs,GIDs}, but it seems got forgot when deleted. :)

Comment (yzgyyang) [reply science/orthanc/Makefile:46]: Oh! Today I learned. Though from what I can see, the UID/GID stayed the same, so it actually works as-is this time, correct?

Comment (lwhsu) [reply science/orthanc/Makefile:46]: Yes, just letting you know this patch doesn't contain UIDs/GIDs is because of a previous fault. :)

**Pull request merged**